docs/content/standards/kiosk-apps/nft-rental.mdx

```diff
@@ -0,0 +1,428 @@
+
+- List for renting
+- Delist from renting
+- Rent
+- Borrow by reference and borrow by value
+- Reclaim for the lender
+
+### Structs
+
+The object model of the `nft_rental` module provides the structure of the app, beginning with the `Rentables` object. The struct only has the `drop` ability and acts as the extension key for the kiosk `Rentables` extension.
+
+```move
+struct Rentables has drop {}
+```
+
+The `Rented` struct represents a rented item. The only field the struct includes is the ID of the object. It is used as the DF key in the borrower's `Bag <key, value>` entry when someone is actively borrowing an item.
+
+```move
+struct Rented has store, copy, drop { 
+    id: ID 
+}
+```
+
+The `Listed` struct represents a listed item. The only field the struct includes is the ID of the object. It is used as the DF key in the renter's `Bag <key, value>` entry after an item is listed for renting. 
+
+```move
+struct Listed has store, copy, drop { 
+    id: ID 
+}
+```
+
+The `Promise` struct is created for borrowing by value. The `Promise` operates as the [hot potato](../../concepts/sui-move-concepts/patterns/hot-potato.mdx) that can only be resolved by returning the item back to the extension's `Bag`.
+The Promise field lacks the `store` ability as it shouldn't be wrapped inside other objects.
+It also lacks the `drop` ability because only the `return_val` function can consume it.
+
+```move
+struct Promise {
+  item_id: Rented,
+  duration: u64,
+  start_date: u64,
+  price_per_day: u64,
+  renter: address,
+  borrower_kiosk: ID
+}
+```
+
+The `Rentable` struct is as a wrapper object that holds an asset that is being rented. Contains information relevant to the rental period, cost, and renter.
+This struct requires the `store` ability because it stores a value `T` that definitely also has `store`.
+
+```move
+struct Rentable<T: key + store> has store {
+      object: T,
+      duration: u64, // total amount of time offered for renting in days
+      start_date: Option<u64>, // initially undefined, is updated once someone rents it
+      price_per_day: u64,
+      kiosk_id: ID // the kiosk id that the object was taken from
+}
+```
+
+The `RentalPolicy` struct is a shared object that every creator mints. The struct defines the royalties the creator receives from each rent invocation.
+
+```move
+struct RentalPolicy<phantom T> has key, store {
+	id: UID,
+	balance: Balance<SUI>,
+	/// Note: Move does not support float numbers.
+	/// If you need to represent a float, you need to determine the desired precision and use a larger integer representation.
+	/// For example, percentages can be represented using basis points:
+	/// 10000 basis points represents 100% and 100 basis points represents 1%.
+	amount_bp: u64
+}
+```
+
+The `ProtectedTP` object is a shared object that creators mint to enable renting, regardless of their intent to enforce royalties. The object provides authorized access to an empty `TransferPolicy`.
```

Comment:
It would be useful to have some detail here on whether this requirement comes from Kiosk, or from this particular usage in renting, or because this is a demo, etc.

Reply:
hmm, I would say this is because of 2 main reasons:
1. Kiosk disallows royalty enforced items to be freely tradable, so by design we have created a work around that bypasses this guarantee solely for the purposes of renting.
2. Kiosk Extensions require a TP to be present when `placing` assets to Kiosks to avoid placing non tradable assets.

Reply:
Okay great -- this definitely clears some things up for me. It would be worth adding more detail to the doc content here on:

The fact that without this construct, if the creator wanted to enforce royalties, Kiosk prevents the free transfer of the asset, and this is to allow that in the case of rentals. The existing passage:

> regardless of their intent to enforce royalties

reads to me like the two concepts are orthogonal, which is slightly different.

And your second bullet helps clarify what the `TransferPolicy` is authorizing (only objects with transfer policies can be placed in Kiosks, even in the case of rentals).

Reply:
Ok will try to explain this further!